Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -372,22 +372,10 @@ class Connection : public ThreadSafeRefCounted<Connection, WTF::DestructionThrea
     size_t pendingMessageCountForTesting() const;
     void dispatchOnReceiveQueueForTesting(Function<void()>&&);
 
-    template<typename T, typename C>
-    static AsyncReplyHandler makeAsyncReplyHandler(C&& completionHandler, ThreadLikeAssertion callThread = CompletionHandlerCallThread::AnyThread)
-    {
-        // FIXME: The above uses AnyThread because the API contract on invalid sends does not make sense.
-        return AsyncReplyHandler {
-            {
-                [completionHandler = WTFMove(completionHandler)] (Decoder* decoder) mutable {
-                    if (decoder && decoder->isValid())
-                        T::callReply(*decoder, WTFMove(completionHandler));
-                    else
-                        T::cancelReply(WTFMove(completionHandler));
-                }, callThread
-            },
-            nextAsyncReplyHandlerID()
-        };
-    }
+    template<typename T, typename C> static AsyncReplyHandler makeAsyncReplyHandler(C&& completionHandler, ThreadLikeAssertion callThread = CompletionHandlerCallThread::AnyThread);
```

Comment:
Can these be `private`?

Reply:
`makeAsyncReplyHandler()` has to be public, it's also used from  MessageSender and AuxiliaryProcessProxy classes. The new ones can be private, so I'll make them.